src/test/kotlin/org/elmPerformanceTests/ElmHighlightingPerformanceTest.kt

```diff
@@ -0,0 +1,103 @@
+/*
+The MIT License (MIT)
+
+Derived from intellij-rust
+Copyright (c) 2015 Aleksey Kladov, Evgeny Kurbatsky, Alexey Kudinkin and contributors
+Copyright (c) 2016 JetBrains
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+ */
+
+package org.elmPerformanceTests
+
+import com.intellij.psi.util.PsiModificationTracker
+import org.elm.lang.core.psi.descendantsOfType
+import org.elm.lang.core.resolve.ElmReferenceElement
+import org.elm.utils.Timings
+
+class ElmHighlightingPerformanceTest : ElmRealProjectTestBase() {
+
+    // It is a performance test, but we will measure performance ourselves
+    override fun isPerformanceTest(): Boolean = false
+
+    fun `test highlighting elm-json-tree-view`() =
+            repeatTest { highlightProjectFile(JSON_TREE_VIEW, "src/JsonTree.elm") }
+
+    fun `test highlighting elm-spa-example`() =
+            repeatTest { highlightProjectFile(SPA, "src/Page/Article/Editor.elm") }
+
+    fun `test highlighting elm-css`() =
+            repeatTest { highlightProjectFile(ELM_CSS, "src/Css.elm") }
+
+    fun `test highlighting elm-css 2`() =
+            repeatTest { highlightProjectFile(ELM_CSS, "src/Css/Transitions.elm") }
+
+    fun `test highlighting elm-list-extra`() =
+            repeatTest { highlightProjectFile(LIST_EXTRA, "src/List/Extra.elm") }
+
+    fun `test highlighting elm-dev-tools`() =
+            repeatTest { highlightProjectFile(DEV_TOOLS, "src/DevTools.elm") }
```

Comment:
Sorry! I wrote this code on the airplane using whatever Elm projects I had checked-out on my laptop over the last couple of years. I'll fix it. Pinning to a specific commit is a good idea.

Reply:
I fixed things up to match the current state of master, but I punted on locking it to a specific commit sha. It would be too error-prone since, without automating the git clone process, the git-ignored `testData` on disk could easily diverge from the test code. I would like to automate it in the future, but for now at least this is a step in the right direction.